src/inspector_profiler.cc

```diff
@@ -78,77 +81,103 @@ static void WriteResult(Environment* env,
   Debug(env, DebugCategory::INSPECTOR_PROFILER, "Written result to %s\n", path);
 }
 
+bool StringViewToUTF8(const v8_inspector::StringView& source,
+                      std::vector<char>* utf8_out,
+                      size_t* utf8_length) {
+  size_t source_len = source.length();
+  if (source.is8Bit()) {
+    const char* latin1 = reinterpret_cast<const char*>(source.characters8());
+    *utf8_length = simdutf::utf8_length_from_latin1(latin1, source_len);
+    utf8_out->resize(*utf8_length);
+    size_t result_len =
+        simdutf::convert_latin1_to_utf8(latin1, source_len, utf8_out->data());
+    return *utf8_length == result_len;
+  }
+
+  const char16_t* utf16 =
+      reinterpret_cast<const char16_t*>(source.characters16());
+  *utf8_length = simdutf::utf8_length_from_utf16(utf16, source_len);
+  utf8_out->resize(*utf8_length);
+  size_t result_len =
+      simdutf::convert_utf16_to_utf8(utf16, source_len, utf8_out->data());
+  return *utf8_length == result_len;
+}
+
 void V8ProfilerConnection::V8ProfilerSessionDelegate::SendMessageToFrontend(
     const v8_inspector::StringView& message) {
   Environment* env = connection_->env();
   Isolate* isolate = env->isolate();
   HandleScope handle_scope(isolate);
   Local<Context> context = env->context();
   Context::Scope context_scope(context);
-
   const char* type = connection_->type();
-  // Convert StringView to a Local<String>.
-  Local<String> message_str;
-  if (!String::NewFromTwoByte(isolate,
-                              message.characters16(),
-                              NewStringType::kNormal,
-                              message.length())
-           .ToLocal(&message_str)) {
-    fprintf(
-        stderr, "Failed to convert %s profile message to V8 string\n", type);
-    return;
-  }
 
   Debug(env,
         DebugCategory::INSPECTOR_PROFILER,
-        "Receive %s profile message\n",
+        "Received %s profile message\n",
         type);
 
-  Local<Value> parsed;
-  if (!v8::JSON::Parse(context, message_str).ToLocal(&parsed) ||
-      !parsed->IsObject()) {
-    fprintf(stderr, "Failed to parse %s profile result as JSON object\n", type);
+  std::vector<char> message_utf8;
+  size_t message_utf8_length;
+  if (!StringViewToUTF8(message, &message_utf8, &message_utf8_length)) {
+    fprintf(
+        stderr, "Failed to convert %s profile message to UTF8 string\n", type);
     return;
   }
+  // Allocate extra padding for JSON parsing.
+  message_utf8.resize(message_utf8_length + simdjson::SIMDJSON_PADDING);
```

Comment:
You could potentially avoid the call to resize by passing a padding parameter to the StringViewToUTF8 function.

Reply:
Good idea, merged the padding into the helper.